### PR TITLE
Vault base setup

### PR DESCRIPTION
**Pull request by glrf**
Wrote a basic ansible role to setup vault and automatically unseal it

### CODE REVIEW

File: ansible/roles/vault/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+
+vault_version: "1.2.3"
```
Comment (bimmlerd) [inline ansible/roles/vault/defaults/main.yml:2]: lol is it really 1.2.3? xD

File: ansible/roles/vault/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+vault_architecture_map:
+  # this first entry seems... redundant (but it's required for reasons)
+  amd64: amd64
+  x86_64: amd64
+  armv7l: arm
+  aarch64: arm64
+vault_architecture: "{{ vault_architecture_map[ansible_architecture] }}"
+vault_os: "{{ ansible_system|lower }}"
+vault_pkg: "vault_{{ vault_version }}_{{ vault_os }}_{{ vault_architecture }}.zip"
+vault_zip_url: "https://releases.hashicorp.com/vault/{{ vault_version }}/vault_{{ vault_version }}_{{ vault_os }}_{{ vault_architecture }}.zip"
```
Comment (bimmlerd) [inline ansible/roles/vault/defaults/main.yml:12]: just curious, why the dl path but then commit the file?

File: ansible/roles/vault/tasks/init.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  register: sealed
+
+- name: Read in key
+  include_vars:
+    file: "~/.vault/key.json"
+    name: key
+  tags: init
+  when: sealed.json.sealed
+
+- name: Unseal
```
Comment (bimmlerd) [inline ansible/roles/vault/tasks/init.yml:49]: slight duplication of unseal

**Review by bimmlerd — APPROVED** [review]
lgtm, if anything only minor stuff

File: ansible/roles/vault/defaults/main.yml (hunk 1 above)
Comment (glrf) [reply ansible/roles/vault/defaults/main.yml:2]: jup that's not an error^^

File: ansible/roles/vault/defaults/main.yml (hunk 2 above)
Comment (glrf) [reply ansible/roles/vault/defaults/main.yml:12]: did not want to commit the file. Is now removed

File: ansible/roles/vault/tasks/init.yml (hunk 3 above)
Comment (glrf) [reply ansible/roles/vault/tasks/init.yml:49]: Fair. But I'm not sure if moving this into a different file to include would actually make the code more readable. I don't expect  to repeat this a third time

**Pull request merged**